packages/frontend/src/components/dialogs/ConfirmDeleteMessage.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useContext } from 'react'
+
+import Dialog, {
+  DialogBody,
+  DialogContent,
+  DialogFooter,
+  FooterActions,
+} from '../Dialog'
+import FooterActionButton from '../Dialog/FooterActionButton'
+import useTranslationFunction from '../../hooks/useTranslationFunction'
+
+import type { DialogProps } from '../../contexts/DialogContext'
+import { BackendRemote, Type } from '../../backend-com'
+import { C } from '@deltachat/jsonrpc-client'
+import { ScreenContext } from '../../contexts/ScreenContext'
+
+export type Props = {
+  accountId: number
+  msg: Type.Message
+  chat: Type.FullChat
+} & DialogProps
+
+export default function ConfirmDeleteMessageDialog(props: Props) {
+  const tx = useTranslationFunction()
+  const { userFeedback } = useContext(ScreenContext)
+
+  const { accountId, msg, chat, onClose } = props
+
+  const deleteForAllPossible =
+    msg.sender.id === C.DC_CONTACT_ID_SELF &&
+    !msg.isInfo &&
+    chat.canSend &&
+    !chat.isSelfTalk
+
+  const deleteMessage = (deleteForEveryone: boolean) => {
+    if (deleteForEveryone) {
+      BackendRemote.rpc
+        .deleteMessagesForAll(accountId, [msg.id])
+        .catch((err: Error) => {
+          userFeedback({
+            type: 'error',
+            text: err.message,
+          })
+        })
+    } else {
+      BackendRemote.rpc
+        .deleteMessages(accountId, [msg.id])
+        .catch((err: Error) => {
+          userFeedback({
+            type: 'error',
+            text: err.message,
+          })
+        })
+    }
+    onClose()
+  }
+
+  return (
+    <Dialog onClose={onClose}>
+      <DialogBody>
+        <DialogContent>
+          <p className='whitespace'>{tx('ask_delete_message')}</p>
+        </DialogContent>
+      </DialogBody>
+      <DialogFooter>
+        <FooterActions>
+          <FooterActionButton
+            styling='secondary'
+            onClick={() => onClose()}
+            data-testid='cancel'
+          >
+            {tx('cancel')}
+          </FooterActionButton>
+          <FooterActionButton
+            styling={'danger'}
+            onClick={() => deleteMessage(false)}
+            data-testid='delete_for_me'
+          >
+            {tx('delete_for_me')}
```

Comment:
Same as in https://github.com/deltachat/deltachat-android/pull/3633/files#diff-6046f8bf3371c6d149fcd32fc2b3ac0e1536a66e34b13e0096208086136529a5R348

```suggestion
            {chat.isSelfTalk ? tx('delete') : tx('delete_for_me')}
```